### PR TITLE
Implement Side Menu + Navigation system

### PR DESCRIPTION
**Pull request by RubyKozel**

### CODE REVIEW

File: src/screens/Navigators.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import {LoadingScreen, NewsFeedScreen, SignInScreen, SignUpScreen, NewPostScreen} from "./Screens";
+import {createStackNavigator} from "react-navigation-stack";
+import {TopHeader} from "../components/TopHeader.js"
+import {MaterialCommunityIcons} from "react-native-vector-icons";
```
Comment (YossiSaadi) [inline src/screens/Navigators.js]: No need to use this, you can use the 'Icon' from 'react-native-elements' and ignoring the all 'react-native-vector-icons' lib as it can also cause problems with Expo.
Expo is already implementing it with '@expo/vector-icons'.
But I think it'll be better to use "< Icon name='someIconName' type='material-community'  / >".

File: src/screens/Navigators.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  screen: createStackNavigator({NewPostScreen}, {
+    mode: 'modal',
+    headerMode: 'float',
+    defaultNavigationOptions: {
+      header: () => <TopHeader icon='arrow-back' title={"NEW POST"}
+                               onPress={(navigation) => navigation.navigate('NewsFeedScreen')}/>
+    }
+  }),
+  navigationOptions: {
+    title: "NEW POST", // TODO: How to use context outside of functional components?
```
Comment (YossiSaadi) [inline src/screens/Navigators.js]: It's not possible, the React team does not allow Hooks outside of functional components, another workaround is needed here if we're using this kind of implementation to Navigator

File: src/screens/NewsFeedScreen.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import {Button, Text, View} from "react-native";
```
Comment (YossiSaadi) [inline src/screens/NewsFeedScreen.js]: Wherever you can use the 'react-native-elements', it will make our life easier (I made a 'theme' implementation, and besides that, their components give us more features).
Here you can use it for Button and Text

File: src/screens/NewPostScreen.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import {Text, View} from "react-native";
+
```
Comment (YossiSaadi) [inline src/screens/NewPostScreen.js]: I implemented this component, don't expand it

File: .gitignore [hunk 5]
```diff
@@ -11,6 +11,7 @@ npm-debug.*
 *.orig.*
 web-build/
 web-report/
+package-lock.json
```
Comment (YossiSaadi) [inline .gitignore]: Why did you add this to ignore?
It is recommended that this file will be on VCS repo

File: package.json [hunk 6]
```diff
@@ -16,14 +16,18 @@
...
     "i18n-js": "^3.5.0",
+    "native-base": "^2.13.8",
     "react": "^16.8.3",
     "react-dom": "^16.8.3",
     "react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
+    "react-native-drawer": "^2.5.1",
     "react-native-elements": "^1.2.6",
     "react-native-gesture-handler": "~1.3.0",
     "react-native-reanimated": "~1.2.0",
+    "react-native-safe-area-context": "^0.6.1",
```
[the remark this answers is not in the document]
Comment (RubyKozel) [reply package.json]: This will be removed

File: src/components/Drawer.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+export const Drawer = (props) => {
+  return (
+    <ScrollView>
+      <SafeAreaView
+        style={styles.container}
+        forceInset={{ top: 'always', horizontal: 'never' }}>
+        <DrawerItems {...props} />
+        <Button title="Logout" onPress={async () => {
+          await AsyncStorage.clear();
```
[the remark this answers is not in the document]
Comment (RubyKozel) [reply src/components/Drawer.js]: I'll remove AsyncStorage.clear(), but if we'll use AsyncStorage ever in the project to save any key-val pairs, it should be cleared on logout.
I'll add the signOut method and call it on the button press

File: src/screens/Navigators.js (hunk 1 above)
Comment (RubyKozel) [reply src/screens/Navigators.js]: Will use <Icon name= .. type= .. >

File: src/screens/Navigators.js (hunk 2 above)
Comment (RubyKozel) [reply src/screens/Navigators.js]: Leaving this as an open question

File: src/screens/NewsFeedScreen.js (hunk 3 above)
Comment (RubyKozel) [reply src/screens/NewsFeedScreen.js]: Good to know! will import components from "react-native-elements" from now on

File: src/screens/NewPostScreen.js (hunk 4 above)
Comment (RubyKozel) [reply src/screens/NewPostScreen.js]: This component is used for the sole purpose of testing the DrawerNavigator, will not add anything to it

File: .gitignore (hunk 5 above)
Comment (RubyKozel) [reply .gitignore]: AFAIN, package-lock.json should be excluded (as it is generated per install). I do see recommendations to add to VCS to prevent inconsistent installs, so I will add it back.
